src/trix/controllers/level_2_input_controller.js

```diff
@@ -73,14 +73,18 @@ export default class Level2InputController extends InputController {
     beforeinput(event) {
       const handler = this.constructor.inputTypes[event.inputType]
 
-      // Handles bug with Siri dictation on iOS 18+.
-      if (!event.inputType) {
-        this.render()
-      }
+      const immmediateRender = shouldRenderInmmediatelyToDealWithIOSDictation(event)
 
       if (handler) {
         this.withEvent(event, handler)
-        this.scheduleRender()
+
+        if (!immmediateRender) {
```

Comment:
I'm conservatively applying the patch to only iOS and only in certain scenarios. It would be simpler to just render things synchronously all the times. We should probably just always render synchronously if more of these show up. Scheduled rendering waits for an animation frame to render, which is good to avoid locking screen rendering, but it's problematic when receiving many events in a very quick sequence.